api/specification.yaml

```diff
@@ -2,118 +2,194 @@ openapi: 3.0.2
-            value: "24.9384"
-        name: origin_lon
-        description: Longitude for trip origin (using WGS 84).
-        schema:
-          format: float
+          format: double
+        longitude:
           type: number
-        in: query
-        required: false
-      - examples:
-          example_lat:
-            value: "60.1699"
-        name: destination_lat
-        description: Latitude for trip destination (using WGS 84).
-        schema:
-          format: float
+          format: double
+      required:
+        - latitude
+        - longitude
+    itinerary_option_origin_destination:
+      type: object
+      description: Itinerary option where the trip origin and destination coordinates are known. Using WGS-84 lat/long.
+      properties:
+        transport_mode:
+          $ref: "#/components/schemas/transport_mode"
+        origin:
+          $ref: "#/components/schemas/coordinate"
+        destination:
+          $ref: "#/components/schemas/coordinate"
+        vehicle_occupancy:
           type: number
-        in: query
-        required: false
-      - examples:
-          example_lon:
-            value: "24.9384"
-        name: destination_lon
-        description: Longitude for trip destination (using WGS 84).
-        schema:
-          format: float
+      required:
+        - transport_mode
+        - origin
+        - destination
+      additionalProperties: true
+      example:
+        origin:
+          latitude: 60.2055
+          longitude: 24.9384
+        destination:
+          latitude: 60.2055
+          longitude: 24.6559
+        transport_mode: "SMALL_CAR"
+        other_property: "something else"
+    itinerary_option_distance_km:
+      type: object
+      description: Itinerary option where the trip distince in kilometers is provided.
+      properties:
+        transport_mode:
+          $ref: "#/components/schemas/transport_mode"
+        distance_km:
           type: number
-        in: query
-        required: false
-      - examples:
-          example_small_car_mode:
-            value: SMALL_CAR
-        name: transport_mode
-        description: Mode of transportation for given trip (leg).
-        schema:
-          enum:
-            - LIGHT_RAIL
-            - SMALL_CAR
-            - LARGE_CAR
-            - SCOOTER
-            - BUS
-            - WALK
-            - BICYCLE
-            - CAR
-            - TRAM
-            - SUBWAY
-            - RAIL
-            - FERRY
-            - CABLE_CAR
-            - GONDOLA
-            - FUNICULAR
-            - TRANSIT
-            - LEG_SWITCH
-            - AIRPLANE
+        vehicle_occupancy:
+          type: number
+      required:
+        - transport_mode
+        - distance_km
+      additionalProperties: true
+      example:
+        transport_mode: "SMALL_CAR"
+        vehicle_occupancy: 2
+        distance_km: 20
+        other_property: "something else"
+    transport_mode:
+      type: string
+      enum:
```

Comment:
Would it be useful to put the source of this in a description property?

Reply:
I guess `WAIT` is not one of the options as it's not relevant to CO2 calculation?
Would clients need to strip such `WAIT` legs out before calling this API?

Reply:
I can add WAIT to the enum.